substrate/client/rpc-servers/src/lib.rs

```diff
@@ -81,26 +82,30 @@ pub async fn start_server<M: Send + Sync + 'static>(
 		max_connections,
 		max_subs_per_conn,
 		metrics,
+		message_buffer_capacity,
 		id_provider,
 		tokio_handle,
 		rpc_api,
 	} = config;
 
-	let host_filter = hosts_filtering(cors.is_some(), &addrs);
+	let std_listener = TcpListener::bind(addrs.as_slice()).await?.into_std()?;
```

Comment:
Changed this to "bind" the port before starting the server in order to 
enable "host filter" for the actual port.